src/core/components/stateful.ts

```diff
@@ -409,6 +413,10 @@ export abstract class StatefulComponentBase extends ComponentBase {
         });
         return config;
     }
+
+    private static _isValueSaveable(value: any): boolean {
```

Comment:
Add a documentation block.

Reply:
Also, perhaps this should be moved to utils as a free function instead of being a method on the component?

Reply:
And it could be named `isValueSerializable`.

Reply:
>And it could be named isValueSerializable

Hm, `{ f: () => {} }` _is_ serializable, it's just that `f` will be omitted.

Reply:
>Also, perhaps this should be moved to utils as a free function instead of being a method on the component?

The function is kinda specific to `StatefulComponentBase` class so I'd leave it where it is.

Reply:
Doesn't it just check if the value is serializable to JSON? So closures aren't really serializable, as you can't include arbitrary code and closure context as JSON without some custom encoding/decoding.

Reply:
No, it checks if e.g. the object contains functions:

```ts
{
    str: 'str', // ok
    func: () => {}, // not ok
}
```

Saved object will lack the `func` property and we want to warn about it.

Reply:
Although hm, `{ prop: undefined }` will also not be allowed.

Reply:
> No, it checks if e.g. the object contains functions:

This is not true, the following will also return false and doesn't contain any functions:
```ts
{
    foo: undefined,
}
```

The body says:
```ts
_.isEqual(JSON.parse(JSON.stringify(value)), value);
```

So it performs a round-trip serialize/deserialize and checks that it is equal to the original value. To me this looks like it checks if the value is serializable. Or to be more precise it checks if the value is serializable in a way that can be reproduced during deserialization.

IMO you can't say that something is serializable just because `JSON.stringify` doesn't crash but omits the value instead. For example, serializing the closure directly, `JSON.stringify` will return `undefined` which seems correct as you cannot serialize a closure. Due to handling of `undefined` values, if a closure is included in an object it will be omitted the same as an actual `undefined` in an object is.

An object that contains a closure somewhere is IMO not serializable if that closure will simply disappear during serialization.

Reply:
>An object that contains a closure somewhere is IMO not serializable if that closure will simply disappear during serialization.

That's true.

Maybe the function should be renamed to `isJSONSerializable`?

Reply:
:+1: Yeah, I think this would sound better.

Reply:
Fixed and updated the changelog.